secure_account/models/account_analytic_line.py

```diff
@@ -40,10 +52,11 @@ def unlink(self):
 
     @api.multi
     def write(self, vals):
-        if PROTECTED_FIELDS.intersection(vals):
+        protected_fields = self.get_protected_fields()
+        if protected_fields.intersection(vals):
             for line in self:
                 if line.move_id.move_id.state == 'posted':
-                    field = tuple(PROTECTED_FIELDS.intersection(vals))[0]
+                    field = tuple(protected_fields.intersection(vals))[0]
```

Comment:
I did not figure that out in the 8.0 version, but the intersection here can be done one time before the for loop.